### ISSUE TITLE
Reply posted via third-party client does not store any metadata alongside comment

### ISSUE THREAD

**Issue by jeherve** [akirk/enable-mastodon-apps#166]
#### Steps to reproduce

1. Start with a site running ActivityPub, Friends, Enable Mastodon Apps.
2. Authenticate onto your site using a third-party client like Tusky.
3. Publish a new post in WP.
4. Let someone else reply to the post on the Fediverse.
5. You'll receive a notification about the reply in Tusky.
6. Within Tusky, hit reply and reply to that person.
7. The reply will be successfully sent, and successfully stored on your site, but no metadata will be attached to the comment. As a result, the comment will not be displayed anywhere on the site, and the person will not receive any notification about your reply.

Here is an example of a comment posted via Tusky:

```
$ wp comment get 31
+----------------------+------------------------------------------------------------+
| Field                | Value                                                      |
+----------------------+------------------------------------------------------------+
| comment_ID           | 31                                                         |
| comment_post_ID      | 431                                                        |
| comment_author       | 1                                                          |
| comment_author_email |                                                            |
| comment_author_url   |                                                            |
| comment_author_IP    |                                                            |
| comment_date         | 2024-08-08 23:36:43                                        |
| comment_date_gmt     | 2024-08-08 21:36:43                                        |
| comment_content      | comment text stored here                                   |
| comment_karma        | 0                                                          |
| comment_approved     | 1                                                          |
| comment_agent        |                                                            |
| comment_type         | comment                                                    |
| comment_parent       | 0                                                          |
| user_id              | 1                                                          |
+----------------------+------------------------------------------------------------+
```

When replying from the site directly instead of via the third-party app, the comment metadata is properly added and the person consequently receives a notification of the reply.

**Comment by akirk** [comment]
The problem here is the `comment_parent == 0` which was not correctly determined. I am close to a fix in #176. Thanks for your patience!

**Comment by jeherve** [comment]
Thank you!
